### PR TITLE
Upgrade scene select to use viirs data

### PR DESCRIPTION
**Pull request by dunkgray**
from - https://gajira.atlassian.net/browse/DSNS-217
Overall need to check if dynamic VIIRS data is available for ARD processing. The check is that for a given date for both the I band and M band date directories are present. The date directories are assumed to be in the form YYYY.MM.DD. Pass in a date to scene select that represents when dynamic VIIRS data is used for BRDF values in ARD processing. Before the date MODIS data is used. Will need a VIIRS data directory for each band too.

The location of the VIIRS data (I band and M band);

/g/data/v10/eoancillarydata-2/BRDF/VNP43IA1.001

/g/data/v10/eoancillarydata-2/BRDF/VNP43MA1.001

### CODE REVIEW

File: tests/test_check_ancillary.py [hunk 1]
```diff
@@ -11,9 +11,49 @@
...
+    af_ob = AncillaryFiles(
+        viirs_m_path=VIIRS_M_TEST_DIR,
+        viirs_i_path=VIIRS_I_TEST_DIR,
+        brdf_dir=BRDF_TEST_DIR,
+        wv_dir=WV_TEST_DIR,
+    )
+    a_dt = datetime.datetime(2020, 8, 2, tzinfo=pytz.UTC)
+    ymd = a_dt.strftime("%Y.%m.%d")
+    ancill_there, msg = af_ob.check_viirs(ymd)
+    assert not ancill_there
```
Comment (gordonyeong) [inline tests/test_check_ancillary.py:42]: i referred to the function, `check_viirs` and in the case of a success, it returns a tuple of (True, '') with the latter being an empty string.
In the case of a failure (ie. no data), a tuple of False, <message> will be returned. The latter , which has a date component indicates there is no VII data for that given date.
 For purposes of conveying that behaviour, I suggest we test add an explicit by checking for an empty string using an `AND`.

 
 `assert ancill_there is False and msg == "", " Unexpectedly, there is brdf vii I data"`

File: tests/test_check_ancillary.py [hunk 2]
```diff
@@ -11,9 +11,49 @@
...
+    af_ob = AncillaryFiles(
+        viirs_m_path=VIIRS_M_TEST_DIR,
+        viirs_i_path=VIIRS_I_TEST_DIR,
+        brdf_dir=BRDF_TEST_DIR,
+        wv_dir=WV_TEST_DIR,
+    )
+    a_dt = datetime.datetime(2020, 8, 3, tzinfo=pytz.UTC)
+    ymd = a_dt.strftime("%Y.%m.%d")
+    ancill_there, msg = af_ob.check_viirs(ymd)
+    assert not ancill_there
```
Comment (gordonyeong) [inline tests/test_check_ancillary.py:54]: Similar suggestion as per the previous.

**Review by gordonyeong — CHANGES_REQUESTED** [review]

File: scene_select/ard_scene_select.py [hunk 3]
```diff
@@ -364,6 +371,9 @@ def l1_filter(
     dc,
     l1_product,
     brdfdir: Path,
+    iviirsdir: Path,
```
[the remark this answers is not in the document]
Comment (dunkgray) [reply scene_select/ard_scene_select.py:374]: Thanks. Updating to i_viirsdir & m_viirsdir

File: tests/ard_reprocessed_l1s/dev_go_reprocess.sh [hunk 4]
```diff
@@ -54,12 +55,15 @@ pkgdir=$basedir/pkgdir$RANDOM
 # a dev run uses the dev database for scene select
 # a prod run uses the prod database for scene select
 # run ['dev'|'prod']
-run='dev'
-#run='prod'
+#run='dev'
+run='prod'
```
[the remark this answers is not in the document]
Comment (dunkgray) [reply tests/ard_reprocessed_l1s/dev_go_reprocess.sh:59]: Thanks. Setting back.

File: tests/test_check_ancillary.py (hunk 1 above)
Comment (dunkgray) [reply tests/test_check_ancillary.py:42]: I've added an assert for the msg too.

File: tests/test_check_ancillary.py (hunk 2 above)
Comment (dunkgray) [reply tests/test_check_ancillary.py:54]: I've added an assert for the msg too.

**Comment by dunkgray** [comment]
ok, I've done fixes. Ready for a re-review.

**Review by gordonyeong — APPROVED** [review]

**Pull request merged**